xenonpy/descriptor/base.py

```diff
@@ -91,6 +92,7 @@ def __init__(
         *,
         on_errors: str = 'raise',
         return_type: str = 'any',
+        target_col: str = None,
```

Comment:
Also can be an integer when no column names were specified

Reply:
It does not support int right now because I do not have a check to say if it is int, use "iloc" instead of loc.
I think your BaseDescriptor also required matching the names, so I presume the same here.
Technically, the column names can also be in other types, like int... and allowing specifying columns with column number not column name will make things very messy, I think. I recommend allowing use of column names only.
I put str to try to encourage people using str for column names... I do not know what would be the right way.
For convenience, maybe change to 
target_col: Union[List[str], str, None] = None,
or
is there something like List[column name]?

Reply:
`target_col: Union[List[str], str, None] = None` is the right way.
I agree with you. We should recommend peoples to use str for cols.